sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -294,24 +318,32 @@ private CollectionCapacityConstraint GetInitializerConstraint(InitializerExpress
                     : CollectionCapacityConstraint.NotEmpty;
             }
 
-            private static ProgramState RemoveCollectionConstraintsFromArguments(InvocationExpressionSyntax invocation,
+            private static ProgramState RemoveCollectionConstraintsFromArguments(ArgumentListSyntax argumentList,
                 ProgramState programState)
             {
-                // Remove the collection constraints from all arguments of the invocation expression
-
-                var values = invocation.ArgumentList.Arguments.Select((node, index) =>
-                {
-                    SymbolicValue value;
-                    programState.ExpressionStack.Pop(out value);
-                    return value;
-                }).ToList();
+                return GetArgumentSymbolicValues(argumentList, programState)
+                    .Aggregate(programState, 
+                        (state, value) => value.RemoveConstraint(CollectionCapacityConstraint.Empty, state));
+            }
 
-                foreach (var value in values)
+            private static IEnumerable<SymbolicValue> GetArgumentSymbolicValues(ArgumentListSyntax argumentList,

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'Microsoft.CodeAnalysis.CSharp.Syntax.BaseArgumentListSyntax' instead of 'Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentListSyntax'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)